libs/app-shell/src/lib/api/extensions/lookup/plugin-descriptor.model.ts

```diff
@@ -1,31 +1,50 @@
-export interface PluginDescriptor {
+export interface AngularIvyComponentDescriptor {
+    /** Uniq name of module in plugin */
     name: string;
-    provider: string
-    version: string;
-    sinceVersion: string;
-    untilVersion: string;
-    remoteEntry: string;
-    remoteName: string;
-    remoteRoute: string;
+    /** Type of a plugin module */
+    type: 'angular-ivy-component';
+    /** Exposed Angular Module */
     exposedModule: string;
-    displayName: string;
+    /** Component name we are planning to inject */
     componentName: string;
-
-    type: Scope;
-    category: string;
-    hasRoutes: boolean;
-    changeNotes: string;
+    /** most likely this property will be removed, but as for this this is a way to register route for a component*/
+    route?: string;
 }
 
+export interface IframePageDescriptor {
```

Comment:
Can you have one common interface PLuginDescription that you have below whcih has common properties and let the others extends it ?

Reply:
This is an interfaces for modules. One container has several exported modules, one uri could host several iframes to inject, etc 
But share same name and version

Reply:
Yes, they definitelly shoudl share some common properties.